src/components/graph-view.js

```diff
@@ -880,15 +880,17 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
           return;
         }
         const sourceNode = sourceNodeById.node;
+        const { canSwapEdge, onSwapEdge } = this.props;
         if (
           edgeEndNode &&
           !this.getEdgeBySourceTarget(draggedEdge.source, edgeEndNode[nodeKey]) &&
-          this.canSwap(sourceNode, edgeEndNode, draggedEdge)
+            this.canSwap(sourceNode, edgeEndNode, draggedEdge)
+            && (!canSwapEdge || canSwapEdge(sourceNode, edgeEndNode, draggedEdge))
```

Comment:
The canSwapEdge check should happen within the this.canSwap function. Please move the check there.